test/functionalTest/cases/2015_notification_templates/notification_templates.test

```diff
@@ -0,0 +1,544 @@
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "04. Create subscription for E1 with URL and QS in template part - QS with substitutions"
+echo "======================================================================================="
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "id" : "E1"
+      }
+    ],
+    "condition": {
+      "attrs": []
+    }
+  },
+  "notification": {
+    "httpExtended": {
+      "url":      "http://127.0.0.1:'${LISTENER_PORT}'/notify",
+      "qs":       { "step": "04", "id": "${id}", "type": "${type}", "a1": "${A1}", "a2": "${A2}", "a3": "${A3}" }
+    }
+  }
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "05. Create subscription for E1 with URL and HEADERS in template part - HEADERS with substitutions"
+echo "================================================================================================="
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "id" : "E1"
+      }
+    ],
+    "condition": {
+      "attrs": []
+    }
+  },
+  "notification": {
+    "httpExtended": {
+      "url":     "http://127.0.0.1:'${LISTENER_PORT}'/notify",
+      "headers": { "entity-id": "${id}", "entity-type": "${type}", "A1": "${A1}", "A2": "${A2}", "A3": "${A3}", "step": "05" },
+      "qs":      { "step": "05" }
+    }
+  }
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "06. Create subscription for E1 with URL, VERB, PAYLOAD, QS and HEADERS in template part - all with substitutions (except VERB, of course)"
+echo "========================================================================================================================================="
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "id" : "E1"
+      }
+    ],
+    "condition": {
+      "attrs": []
+    }
+  },
+  "notification": {
+    "httpExtended": {
+      "url":      "http://127.0.0.1:'${LISTENER_PORT}'/${NOTIFY_URL}",
+      "method":   "DELETE",
+      "payload":  "{ \"A1\": \"${A1}\", \"A2\": \"${A2}\", \"A3\": \"${A3}\" }",
+      "qs":       { "step": "06", "id": "${id}", "type": "${type}", "a1": "${A1}", "a2": "${A2}", "a3": "${A3}" },
+      "headers":  { "entity-id": "${id}", "entity-type": "${type}", "A1": "${A1}", "A2": "${A2}", "A3": "${A3}", "step": "06" }
+    }
+  }
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "07. Create subscription for E1 with PAYLOAD, QS and HEADERS with unknown attribute 'A4' for substitution"
+echo "========================================================================================================"
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "id" : "E1"
+      }
+    ],
+    "condition": {
+      "attrs": []
+    }
+  },
+  "notification": {
+    "httpExtended": {
+      "url":      "http://127.0.0.1:'${LISTENER_PORT}'/notify",
+      "payload":  "{ \"A1\": \"${A1}\", \"A2\": \"${A2}\", \"A4\": \"${A4}\" }",
```

Comment:
Not sure if this included in another .test, but I'd suggest to add also test different from JSON. In particular:

* A test focused in a XML case
* A test focused in a test plain case

Probably not as complete as this .test, but two little .test (one for each case) would be great.